### DISCUSSION TITLE
Is there any way to add attributes to Observable Gauge and Observable Counter types.

### DISCUSSION THREAD

**Discussion by manu-balakrishnan** [open-telemetry/opentelemetry-dotnet#3595] [Q&A]
Hi,

     We are able to create opentelemetry metrics of Observable Gauge and Observable counter types in .net applications however unable to add attributes to it. Is there any way to add attributes to these types of opentelemtry metrics in .net ?

**Comment by alanwest** [comment]
We have an example demonstrating both `ObservableGauge` and `ObservableCounter` here https://github.com/open-telemetry/opentelemetry-dotnet/blob/3af5a3b0b44f16415badb52d38d96f14ef9644a5/docs/metrics/learning-more-instruments/Program.cs#L35-L37

In both cases the callback that generates the measurements adds a `ProcessId` and `ThreadId` attribute. https://github.com/open-telemetry/opentelemetry-dotnet/blob/3af5a3b0b44f16415badb52d38d96f14ef9644a5/docs/metrics/learning-more-instruments/Program.cs#L54-L60
